OSDOCS-10530 OCP 4.15.13 Release Notes (#75800)
openshift/openshift-docs: release_notes/ocp-4-15-release-notes.adoc

For any {product-title} release, always review the instructions on xref:../updating/updating_a_cluster/updating-cluster-web-console.adoc#updating-cluster-web-console[updating your cluster] properly.
====

[id="ocp-4-15-13"]
=== RHSA-2024:2773 - {product-title} 4.15.13 bug fix and security update

Issued: 2024-05-15

{product-title} release 4.15.13, which includes security updates, is now available. The list of bug fixes that are included in the update is documented in the link:https://access.redhat.com/errata/RHSA-2024:2773[RHSA-2024:2773] advisory. The RPM packages that are included in the update are provided by the link:https://access.redhat.com/errata/RHSA-2024:2776[RHSA-2024:2776] advisory.

Space precluded documenting all of the container images for this release in the advisory.

You can view the container images in this release by running the following command:

[source,terminal]
----
$ oc adm release info 4.15.13 --pullspecs
----

[id="ocp-4-15-13-bug-fixes"]
==== Bug fixes

> * Previously, the name of the Security Context Constraint (SCC) was incorrect so there was no functioning built-in cluster role. With this release, the name was changed to `hostmount-anyuid` and the SCC now has a functioning built-in cluster role.  (link:https://issues.redhat.com/browse/OCPBUGS-33277[*OCPBUGS-33277*])

Reviewer: ```suggestion
* Previously, the name of the Security Context Constraint (SCC) was incorrect so there was no functioning built-in cluster role. With this release, the name was changed to `hostmount-anyuid` and the SCC now has a functioning built-in cluster role. (link:https://issues.redhat.com/browse/OCPBUGS-33277[*OCPBUGS-33277*])
```

* Previously, the Ironic Python Agent (IPA) failed when trying to wipe disks because it expected the wrong byte sector size, which caused the node provisioning to fail. With this release, the IPA checks the disk sector size and node provisioning succeeds. (link:https://issues.redhat.com/browse/OCPBUGS-33133[*OCPBUGS-33133*])

> * Previously, Static Persistent Volumes in Azure File on Workload Identity clusters could not be configured due to a bug in the driver causing volume mounts to fail. With this release, the driver has now been fixed and Static Persistent Volumes mount correctly. (link:https://issues.redhat.com/browse/OCPBUGS-33038[*OCPBUGS-33038*])

Reviewer: ```suggestion
* Previously, Static Persistent Volumes in Azure File on Workload Identity clusters could not be configured due to a bug in the driver causing volume mounts to fail. With this release, the driver has now been fixed, and Static Persistent Volumes mount correctly. (link:https://issues.redhat.com/browse/OCPBUGS-33038[*OCPBUGS-33038*])
```

* Previously, during {product-title} updates in performance-tuned clusters, resuming a `MachineConfigPool` resource resulted in additional restarts for nodes in the pool. This was due to the performance profile controller reconciling against outdated machine configurations while the pool was paused. With this update, the controller reconciles against the latest planned machine configurations before the pool resumes, preventing additional node reboots. (link:https://issues.redhat.com/browse/OCPBUGS-32978[*OCPBUGS-32978*])

> * Previously, the load balancing algorithm did not differentiate between active and inactive services when determining weights, and it employed the "random" algorithm excessively in environments with many inactive services or environments routing backends with weight 0. This led to increased memory usage and a higher risk of excessive memory consumption. With this release, changes are made to optimize traffic direction towards active services only and prevent unnecessary use of the "random" algorithm with higher weights, reducing the potential for excessive memory consumption. (link:https://issues.redhat.com/browse/OCPBUGS-32977[*OCPBUGS-32977*])

Reviewer: ```suggestion
* Previously, the load balancing algorithm did not differentiate between active and inactive services when determining weights, and it employed the `random` algorithm excessively in environments with many inactive services or environments routing backends with weight 0. This led to increased memory usage and a higher risk of excessive memory consumption. With this release, changes are made to optimize traffic direction towards active services only and prevent unnecessary use of the `random` algorithm with higher weights, reducing the potential for excessive memory consumption. (link:https://issues.redhat.com/browse/OCPBUGS-32977[*OCPBUGS-32977*])
```

> * Previously, the load balancing algorithm did not differentiate between active and inactive services when determining weights, and it employed the "random" algorithm excessively in environments with many inactive services or environments routing backends with weight 0. This led to increased memory usage and a higher risk of excessive memory consumption. With this release, changes are made to optimise traffic direction towards active services only and prevent unnecessary use of the "random" algorithm with higher weights, which reduces the potential for excessive memory consumption. (link:https://issues.redhat.com/browse/OCPBUGS-32977[*OCPBUGS-32977*])

Reviewer 2: ```suggestion
* Previously, the load balancing algorithm did not differentiate between active and inactive services when determining weights, and it employed the "random" algorithm excessively in environments with many inactive services or environments routing backends with weight 0. This led to increased memory usage and a higher risk of excessive memory consumption. With this release, changes are made to optimize traffic direction towards active services only and prevent unnecessary use of the "random" algorithm with higher weights, reducing the potential for excessive memory consumption. (link:https://issues.redhat.com/browse/OCPBUGS-32977[*OCPBUGS-32977*])
```

> * Previously, If a user created a ContainerRuntimeConfig resource as an extra manifest for a single-node {product-title} cluster (SNO) installation the boostrap process would fail with the error: `more than one ContainerRuntimeConfig found that matches MCP labels`. With this release, the incorrect processing of ContainereRuntimeConfig resources is fixed, which resolves the issue. (link:https://issues.redhat.com/browse/OCPBUGS-30152[*OCPBUGS-30152*])

Reviewer 2: ```suggestion
* Previously, If a user created a `ContainerRuntimeConfig` resource as an extra manifest for a single-node {product-title} cluster (SNO) installation, the boostrap process failed with the error: `more than one ContainerRuntimeConfig found that matches MCP labels`. With this release, the incorrect processing of `ContainerRuntimeConfig` resources is fixed, which resolves the issue. (link:https://issues.redhat.com/browse/OCPBUGS-30152[*OCPBUGS-30152*])
```

* Previously, the Helm Plugin index view did not display the same number of charts as the Helm CLI if the chart names varied. With this release, the Helm catalog now looks for `charts.openshift.io/name` and `charts.openshift.io/provider` so that all versions are grouped together in a single catalog title. (link:https://issues.redhat.com/browse/OCPBUGS-32716[*OCPBUGS-32716*])

> * Previously, the description of the hosted control plane CLI flag `api-server-address` was unclear. With this release the description has been updated for clarity and completeness. (link:https://issues.redhat.com/browse/OCPBUGS-25858[*OCPBUGS-25858*])

Reviewer: ```suggestion
* Previously, the description of the hosted control plane CLI flag `api-server-address` was unclear. With this release, the description has been updated for clarity and completeness. (link:https://issues.redhat.com/browse/OCPBUGS-25858[*OCPBUGS-25858*])
```
